step_function() gives a compiler error unless static casting DataVector expression

# Bug reports:

### Expected behavior:
~~~~
// Add these lines at line 335 of Test_DataVector.cpp
// These lines work
DataVector debug{{-4.0, 0.0, 4.0}};
DataVector debug_combo = -4.0 * debug + 4.0;
DataVector step = step_function(debug);
DataVector step_combo = step_function(debug_combo);
DataVector step_combo2 = step_function(static_cast<DataVector>(-4.0 * debug + 4.0));

// Expect this line to also work (compile and evaluate correctly)
DataVector step_combo3 = step_function(-4.0 * debug + 4.0);
~~~~
### Current behavior:
~~~~
// This line gives a compiler error: 
// spectre/src/Utilities/PointerVector.hpp:37:24: error: 
// type 'const blaze::SIMDf64FmaddExpr<blaze::SIMDdouble, 
// blaze::SIMDdouble, blaze::SIMDdouble>' does not provide 
// a subscript operator
DataVector step_combo3 = step_function(-4.0 * debug + 4.0);
~~~~
### Environment:

Docker

# Feature request:

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Desired feature:

- [ ] Detail 1
- [ ] Detail 2
- [ ] Detail 3

### Detailed discussion:

--- a/src/Utilities/PointerVector.hpp
+++ b/src/Utilities/PointerVector.hpp
@@ -27,19 +27,26 @@ template <typename T>
 BLAZE_ALWAYS_INLINE SIMDdouble step_function(const SIMDf64<T>& v) noexcept
 #if BLAZE_AVX512F_MODE || BLAZE_MIC_MODE
 {
-  return _mm512_set_pd((~v)[7] < 0.0 ? 0.0 : 1.0, (~v)[6] < 0.0 ? 0.0 : 1.0,
-                       (~v)[5] < 0.0 ? 0.0 : 1.0, (~v)[4] < 0.0 ? 0.0 : 1.0,
-                       (~v)[3] < 0.0 ? 0.0 : 1.0, (~v)[2] < 0.0 ? 0.0 : 1.0,
-                       (~v)[1] < 0.0 ? 0.0 : 1.0, (~v)[0] < 0.0 ? 0.0 : 1.0);
+  return _mm512_set_pd((~v).eval().value[7] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[6] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[5] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[4] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[3] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[2] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[1] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[0] < 0.0 ? 0.0 : 1.0);
 }
 #elif BLAZE_AVX_MODE
 {
-  return _mm256_set_pd((~v)[3] < 0.0 ? 0.0 : 1.0, (~v)[2] < 0.0 ? 0.0 : 1.0,
-                       (~v)[1] < 0.0 ? 0.0 : 1.0, (~v)[0] < 0.0 ? 0.0 : 1.0);
+  return _mm256_set_pd((~v).eval().value[3] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[2] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[1] < 0.0 ? 0.0 : 1.0,
+                       (~v).eval().value[0] < 0.0 ? 0.0 : 1.0);
 }
 #elif BLAZE_SSE2_MODE
 {
-  return _mm_set_pd((~v)[1] < 0.0 ? 0.0 : 1.0, (~v)[0] < 0.0 ? 0.0 : 1.0);
+  return _mm_set_pd((~v).eval().value[1] < 0.0 ? 0.0 : 1.0,
+                    (~v).eval().value[0] < 0.0 ? 0.0 : 1.0);
 }
 #else
 {

--- a/tests/Unit/DataStructures/Test_DataVector.cpp
+++ b/tests/Unit/DataStructures/Test_DataVector.cpp
@@ -296,6 +296,11 @@ void test_datavector_math() noexcept {
   const DataVector step_function_data{-12.3, 2.0, -4.0, 0.0, 7.0, -8.0};
   check_vectors(step_function(wrap<WrapLeftOp>(step_function_data)),
                 DataVector{0.0, 1.0, 0.0, 1.0, 1.0, 0.0});
+  // check that the step_function works with multiple blaze operations
+  // prior to step evaluation
+  check_vectors(step_function(wrap<WrapLeftOp>(0.5 * step_function_data
+                                               + DataVector(6, 1.0))),
+                DataVector{0.0, 1.0, 0.0, 1.0, 1.0, 0.0});
 
   check_vectors(sqrt(wrap<WrapLeftOp>(nine)), DataVector(num_pts, 3.0));
   check_vectors(invsqrt(wrap<WrapLeftOp>(nine)),

